hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadataBuilder.java

```diff
@@ -106,12 +135,14 @@ public void addNonPayerKey(final AccountID id) {
      * @param id given accountId
      * @param failureStatus given failure status
      */
-    public void addNonPayerKey(final AccountID id, final ResponseCodeEnum failureStatus) {
+    public SigTransactionMetadataBuilder addNonPayerKey(
+            final AccountID id, final ResponseCodeEnum failureStatus) {
         if (isNotNeeded(id)) {
```

Comment:
It appears to me, that these are two different scenarios. If `id` is the payer, we can skip the rest, because the payer is added elsewhere. But If `status` is not `OK`, it is wrong to call this method and we should throw an exception, no?

Reply:
No, the intent of failureStatus is to set the given status if there is failure in lookup. Whenever there is a failure in lookup we always give default status of `INVALID_ACCOUNT_ID`. But there are cases where we lookup `transferAccountID`, `allowanceOwner` for which we should set a different status from the default one. In that scenario we use this method.

To be clear I will change `failureStatus` to `failureStatusToUse` ?

Reply:
Sorry for not being clear, I meant something else. In `isNotNeeded()` we do different checks. But some of the conditions in `isNotNeeded()` look like error conditions to me and therefore I think we should throw an `Exception` instead of silently ignoring these calls. For example if you try to add a non-payer key after a failure was already logged (`status != OK`), something is wrong.

Reply:
As discussed , we will need to capture the status to re-lookup during handle.